File: Backend_TFG/src/main/java/com/gert/tfgdam/repository/LineaVentaRepository.java
package com.gert.tfgdam.repository;

import com.gert.tfgdam.entity.LineaVenta;

import java.util.List;

import org.springframework.data.jpa.repository.JpaRepository;

public interface LineaVentaRepository extends JpaRepository<LineaVenta, Long> {
    List<LineaVenta> findByVentaId(Long ventaId);
}

File: Frontend_TFG/app/src/main/java/com/gert/tfgdam/repository/LineaVentaRepository.kt
package com.gert.tfgdam.repository

import com.gert.tfgdam.api.LineaVentaApi
import com.gert.tfgdam.model.LineaVenta
import com.gert.tfgdam.network.RetrofitClient
import retrofit2.Response

class LineaVentaRepository {

    private val api: LineaVentaApi =
        RetrofitClient.instance.create(LineaVentaApi::class.java)

    suspend fun getAll(): Response<List<LineaVenta>> {
        return api.getAll()
    }

    suspend fun getPorId(id: Long): Response<LineaVenta> {
        return api.getPorId(id)
    }

    suspend fun create(lineaVenta: LineaVenta): Response<LineaVenta> {
        return api.create(lineaVenta)
    }

    suspend fun update(id: Long, lineaVenta: LineaVenta): Response<LineaVenta> {
        return api.update(id, lineaVenta)
    }

    suspend fun delete(id: Long): Response<Unit> {
        return api.delete(id)
    }
}